metals/src/main/scala/scala/meta/internal/metals/watcher/ProjectFileWatcher.scala

```diff
@@ -99,7 +99,10 @@ object ProjectFileWatcher {
           !buildTargets.checkIfGeneratedSource(path.toNIO)
 
       if (shouldBeWatched) {
-        if (buildTargets.isSourceFile(path))
+        // buildTargets.isSourceFile(path) was returning false for a bunch of files (not directories), and we were putting them in directories instead
```

Comment:
That's weird. I'm not sure why they were doing this.

Reply:
Yeah, I'm nervous they just meant to partition by something other than files and directories and they just didn't rename `PathsToWatch`, but afaict this is working fine too.